feature-libs/user/account/components/guards/login-as-guest.guard.ts

```diff
@@ -22,10 +24,15 @@ export class LoginAsGuestGuard {
   protected windowRef = inject(WindowRef);
   protected router = inject(Router);
   protected semanticPathService = inject(SemanticPathService);
+  protected authConfigService = inject(AuthConfigService);
 
   canActivate(): Observable<GuardResult> {
     if (
-      this.featureConfigService.isEnabled('authorizationCodeFlowByDefault') &&
+      this.featureConfigService.isEnabled(
+        'nativeSupportForRedirectingOAuthFlows'
+      ) &&
+      this.authConfigService.getOAuthFlow() !==
+        OAuthFlow.ResourceOwnerPasswordFlow &&
```

Comment:
This change is related to the enhancements we made to support oAuth flows involving a redirect.  

We want to disable this new guard if the enhanced support is disabled, or if the oauth flow is a redirecting flow.